packages/ember-headless-form/src/components/headless-form.gts

```diff
@@ -198,10 +198,56 @@ export default class HeadlessFormComponent<
   /**
    * A copy of the passed `@data` stored internally, which is only passed back to the component consumer after a (successful) form submission.
    */
-  internalData: DATA =
-    this.args.dataMode == 'mutable' && this.args.data
-      ? this.args.data
-      : (new TrackedObject(this.args.data ?? {}) as DATA);
+  internalData: DATA = new TrackedObject({}) as DATA;
+
+  @cached
+  get effectiveData(): DATA {
+    const obj: DATA = this.args.data ?? ({} as DATA);
+
+    if (this.args.dataMode == 'mutable') {
```

Comment:
Super nit, but do we want strict equal here?

```suggestion
    if (this.args.dataMode === 'mutable') {
```

Reply:
Oh, absolutely, good catch! 

Shouldn't ESLint have catched that though? 🤔